### ISSUE TITLE
spider object not in shell

### ISSUE THREAD

**Issue by aplicacionamedida** [scrapy/scrapy#920]
In the file shell.py I propose a change in this function "inspect_response":

``` python
def inspect_response(response, spider):
    """Open a shell to inspect the given response"""
    Shell(spider.crawler).start(response=response)
```

This is my proposal:

``` python
def inspect_response(response, spider):
    """Open a shell to inspect the given response"""
    Shell(spider.crawler).start(response=response, spider=spider)
```

Rationale: Without this change, when you call to the function inspect_response(), you obtain a None object inside the "spider" variable. None is the default value for the argument "spider" of the "start()" method of the Shell class.

**Comment by redapple** [comment]
I am able to reproduce this with scrapy 1.1.2

```
$ cat httpbin/spiders/httpbinorg.py
# -*- coding: utf-8 -*-
import scrapy


class HttpbinorgSpider(scrapy.Spider):
    name = "httpbinorg"
    allowed_domains = ["httpbin.org"]
    start_urls = (
        'https://httpbin.org/ip',
    )

    def parse(self, response):
        from scrapy.shell import inspect_response
        inspect_response(response, self)

$ scrapy crawl httpbinorg
2016-09-16 12:59:42 [scrapy] INFO: Scrapy 1.1.2 started (bot: httpbin)
2016-09-16 12:59:42 [scrapy] INFO: Overridden settings: {'BOT_NAME': 'httpbin', 'NEWSPIDER_MODULE': 'httpbin.spiders', 'SPIDER_MODULES': ['httpbin.spiders']}
(...)
2016-09-16 12:59:42 [scrapy] INFO: Spider opened
2016-09-16 12:59:42 [scrapy] INFO: Crawled 0 pages (at 0 pages/min), scraped 0 items (at 0 items/min)
2016-09-16 12:59:42 [scrapy] DEBUG: Telnet console listening on 127.0.0.1:6023
2016-09-16 12:59:42 [scrapy] DEBUG: Crawled (200) <GET https://httpbin.org/ip> (referer: None)
2016-09-16 12:59:43 [root] DEBUG: Using default logger
2016-09-16 12:59:43 [root] DEBUG: Using default logger
[s] Available Scrapy objects:
[s]   crawler    <scrapy.crawler.Crawler object at 0x7f78c1f4b0f0>
[s]   item       {}
[s]   request    <GET https://httpbin.org/ip>
[s]   response   <200 https://httpbin.org/ip>
[s]   settings   <scrapy.settings.Settings object at 0x7f78bd620940>
[s] Useful shortcuts:
[s]   shelp()           Shell help (print this help)
[s]   view(response)    View response in a browser
In [1]: (spider,)
Out[1]: (None,)

In [2]: crawler.spider
Out[2]: <HttpbinorgSpider 'httpbinorg' at 0x7f78bbe8c080>
```

**Comment by elacuesta** [comment]
Looks like this was inadvertently fixed by #2812.

**Comment by kmike** [comment]
👍
